server/src/main/scala/org/http4s/server/Server.scala

```diff
@@ -16,6 +19,25 @@ abstract class Server[F[_]](implicit F: Effect[F]) {
 
   def address: InetSocketAddress
 
+  def isSecure: Boolean
+
+  def baseUri: Uri = Uri(
```

Comment:
This is why our URI type needs reform. ☣️

Reply:
Ugh.